docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
+ 
+ The following sequence diagram below shows how the add feature works. The details of the adding item's values
+ are shown in a separate sequence diagram below:
+ 
+ ![alt text](images/AddFeature.png)
+ 
+ ![alt text](images/AddFeature_SD.png)
+ 
+#### 2.1.2 Design considerations
+
+##### Aspect: Data structure to support the add feature
+
+- Alternative 1 (current choice): User must provided a description for item, Duplicates are
+                                  not allowed in the list. 
+  - Pros: User has minimal potential to see unreasonable list in the Shopping List. For 
+  example, having a item that has only price and quantity but without description and also
+  a list that one item is recorded multiple times in the list.
+  
+  - Cons: Will significantly increase the code base as there is a need to check for the 
+  presence of the variable in user input to avoid duplication, not human friendly in certain 
+  scenario (eg: user wants to have duplicates because the item is for different occasion and 
+  the user wants to record down twice without any elaboration).
+
+
+- Alternative 2: Require user to provide all three values to successfully add the item into 
+                 the list. Duplicates are allowed
+
+  - Pros: User will have a neat and unity Shopping list, less deal with parameter (because users are
+  forced to give all three variables). And the duplicates item are useful in certain specific condition.
+  
+  - Cons: User flexibility will decrease, because user must input all parameters even if he/she does not want to 
+  provide certain variables such as price and quantity, which will result unsuccessful adding items into the list. 
+  Also, duplicate items may confused the user, even though in some specific scenario, duplicate item may be useful to 
+  the user.  
+ 
+
+&nbsp;
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+
+&nbsp;
+
+<!-- @@author trishaangelica --> 
+### 3.2 Edit feature
+#### 3.2.1 Current implementation
+
+The edit feature is implemented using an <code>EditCommand</code> class. This class extends from the main
+<code>Command</code> class. The <code>item</code> object to be edited is identified by the index number provided 
+in the user input. In addition to the index no. , the user input **must also contain at least one** of these parameters: 
+*description*, *price*, *quantity*. 
+
+The process of object creation is as follows:
+
+1. <code>Duke</code> class receives user input from the <code>Ui</code> class. 
+2. A <code>Parser</code> object is created to call its <code>parseCommand</code> method.
+* The <code>Parser</code> object instantiates an <code>EditCommand</code> object based on the user input.
+3. The <code>Duke</code> class calls the <code>execute</code> method of the <code>EditCommand</code> object.
+4. In the <code>execute</code> function, the <code>item</code> to be edited (based on the specified index of the 
```

Comment:
Perhaps, you can state what function you are calling to get the specified index as seen in your diagram for more user readability?

Reply:
thanks for that suggestion, I will include it in!